Source/WebCore/html/HTMLSummaryElement.cpp

```diff
@@ -98,10 +100,30 @@ bool HTMLSummaryElement::isActiveSummary() const
 
 static bool isClickableControl(EventTarget* target)
 {
-    if (!is<Element>(target))
+    if (!is<Element>(target) || is<HTMLSummaryElement>(target))
         return false;
+    if (is<SVGAElement>(target))
+        return true;
     auto& element = downcast<Element>(*target);
-    return is<HTMLFormControlElement>(element) || is<HTMLFormControlElement>(element.shadowHost());
+    if (is<HTMLElement>(element)) {
+        if (downcast<HTMLElement>(element).isInteractiveContent())
+            return true;
+        for (auto& ancestor : ancestorsOfType<HTMLElement>(element)) {
+            if (is<HTMLSummaryElement>(ancestor))
+                return false;
+            if (ancestor.isInteractiveContent())
+                return true;
+        }
```

Comment:
You can use `lineageOfType` instead of having duplicate checks for the current element and the ancestors.